samcli/lib/test_runner/invoke_testsuite.py

```diff
@@ -0,0 +1,140 @@
+from typing import *
+import click
+from samcli.lib.utils.boto_utils import BotoProviderType
+from samcli.commands.exceptions import ReservedEnvironmentVariableException
+
+
+def get_subnets(boto_client_provider: BotoProviderType) -> List[str]:
+    """
+    Queries describe-subnets to get subnets associated with the customer account.
+
+    NOTE: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSubnets.html
+
+    Parameters
+    ----------
+    boto_client_provider : BotoProviderType
+        Provides a boto3 client in order to query the describe-subnets API
+
+    Returns
+    -------
+    List[str]
+        A list of subnet ids associated with a customer account.
+
+    Raises
+    ------
+    botocore.ClientError
+        If the describe_subnets call fails
+    """
+    ec2_client = boto_client_provider("ec2")
+    subnets = ec2_client.describe_subnets().get("Subnets")
+    return [subnet["SubnetId"] for subnet in subnets]
+
+
+def invoke_testsuite(
+    boto_client_provider: BotoProviderType,
```

Comment:
How does the caller specify the region?

Reply:
Caller wouldn't specify the region directly, but would have the region and profile through context, and then construct a `boto_client_provider` to pass. Why would we need region passed directly?